### ISSUE TITLE
Matmul substitution failed when conducting INT8 PTQ for DAMO-YOLO

### ISSUE THREAD

**Issue by hglee98** [sony/model_optimization#1125]
### Issue Type

Bug

### Source

pip (model-compression-toolkit)

### MCT Version

2.1.0

### Describe the issue

As I'm testing the DAMO-YOLO nano model can be deployed to IMX500, I download the onnx file for the `DAMO-YOLO-Nm` from [the official GitHub repository](https://github.com/tinyvision/damo-yolo?tab=readme-ov-file) and conduct MCT PTQ. But I've got an error as described in the title above. I'm guessing the issue is coming from a MatMul layer (<14196x8> <8x1> => <14196x1>). Can it be solved when I substitute the original matmul layer with another working the same as the original? Or Is there any way to fix the issue from the MCT itself? 
Thank you for the time to read this request!

### Code to reproduce the issue

```shell
import model_compression_toolkit as mct
import tensorflow as tf 
from get_calibration_data import representative_dataset_gen
model_path = "./saved_model/damoyolo_tinynasL18_Nm" 

model = tf.keras.models.load_model(model_path+"_float32.h5", compile=False)

tpc = mct.get_target_platform_capabilities(
    fw_name="tensorflow",
    target_platform_name="imx500",
)

mp_config = mct.core.MixedPrecisionQuantizationConfig(num_of_images=5, use_hessian_based_scores=False)
config = mct.core.CoreConfig(mixed_precision_config=mp_config, 
                             quantization_config=mct.core.QuantizationConfig(shift_negative_activation_correction=True))
# Define target Resource Utilization for mixed precision weights quantization (76% of 'standard' 8bits quantization)
resource_utilization_data = mct.core.keras_resource_utilization_data(model,
                                   representative_dataset_gen,
                                   config,
                                   target_platform_capabilities=tpc) <= error here
```

### Log output

```shell
Exception: Matmul substitution failed: Unable to locate weight for node tf.linalg.matmul_1.
```

**Comment by elad-c** [comment]
Hi @hglee98 ,
I see you downloaded an onnx file of the DAMO-YOLO nano model, but in the code you use an H5 keras file. Can you share the code for converting the model from ONNX to H5?

**Comment by hglee98** [comment]
Hello @elad-c, Thank you for your reply! 
I use onnx2tf==1.22.3 to convert onnx to H5. 
You can download the library via this github repository.
https://github.com/PINTO0309/onnx2tf

**Comment by elad-c** [comment]
Thanks,
Can you share your Keras and ONNX versions?

**Comment by hglee98** [comment]
Here's the list of my setup as follows:
python=3.10.14
keras==2.14.0
onnx==1.15.0

**Comment by elad-c** [comment]
Hi @hglee98 ,

Running the code below in an environment with the above versions didn't reproduce the error. There was no error in `mct-nightly` as well.

I converted the onnx file to keras like this:
`onnx2tf -i "damoyolo_tinynasL18_Nm.onnx" --output_h5`

Please note that using conversion tools such `onnx2tf` might generate unusual model graphs which in turn converts to sub-optimal models (e.g. representing `swish` activations as separate `Sigmoid`-`Mul` nodes, or consecutive `reshape` & `transpose` operations to handle the CHW to HWC conversion).




```
k_model = tf.keras.models.load_model("damoyolo_tinynasL18_Nm_float32.h5")

tpc = mct.get_target_platform_capabilities(fw_name="tensorflow", target_platform_name="imx500")


def rep_ds():
    for _ in range(5):
        yield np.random.random((1, 416, 416, 3)).astype(np.float32)


mp_config = mct.core.MixedPrecisionQuantizationConfig(num_of_images=5, use_hessian_based_scores=False)
config = mct.core.CoreConfig(mixed_precision_config=mp_config,
                             quantization_config=mct.core.QuantizationConfig(shift_negative_activation_correction=True))
# Define target Resource Utilization for mixed precision weights quantization (76% of 'standard' 8bits quantization)
resource_utilization_data = mct.core.keras_resource_utilization_data(k_model,
                                   rep_ds,
                                   config,
                                   target_platform_capabilities=tpc)

```

Environment creation in colab:
```
!pip install -q tensorflow==2.14
!pip install -q model-compression-toolkit==2.1.0
!pip install -q onnx==1.15.0
!pip install -q onnx-graphsurgeon
!pip install -q sng4onnx
!pip install -q onnx2tf==1.22.3
```

**Comment by hglee98** [comment]
Thank you for investigating this issue. I appreciate your thorough review. As the problem doesn't seem to be replicable on your end, I'll conduct further investigations in my own environment using the install command that you suggest. I'll close this issue for now, and Thank you again for your time and assistance👍
